src/main/java/Request/RequestParser.java

```diff
@@ -49,6 +50,15 @@ private RequestLine parseRequestLine() throws BadRequestException {
     }
   }
 
+  private String extractMethod(String givenMethod, HashMap<String, String> headers) {
```

Comment:
Does this have to do with cucumber not allowing you to use the PATCH method?

Reply:
It's the `HTTPUrlConnection` class that doesn't allow PATCH. I could probably use another library, but the fix was relatively small.